docs/guides/limitations.md

```diff
@@ -0,0 +1,17 @@
+---
+title: "Limitations"
+linkTitle: "Limitations"
+weight: 998
+slug: limitations
+---
+
+## Tenant ID naming
+
+The tenant ID (also called "user ID" or "org ID") shouldn't include the following characters:
+
+- `/`
```

Comment:
**thought:** Beyond these two characters I think we should consider the built-in restrictions for characters in S3 listed [here](https://docs.aws.amazon.com/AmazonS3/latest/dev/UsingMetadata.html#object-keys). I haven't taken the time to verify which of the special characters listed are actually a problem and restrictions unique to other backends. 

Also it may be worth noting the 1024 character limit puts a max size on the tenant ID for blocks:

`<tenant-id>` + `/` + `<block ULID -- 26 chars>` + `/chunks/` + `<chunk ID -- 6 chars>`

1024 - 1 - 26 - 8 - 6 = 1024 - 41 = 983 maximum length of tenant ID for blocks storage

The same max length as above applies to the rules API. However, rulegroup names/namespaces are also unbounded right now so the max length is more about the interplay between the three. I will add it to a separate issue but it may make sense to have limits configured on each of these things that keep the max length in the sane category so users can't accidentally create objects to big to store in S3.

Reply:
Good point. Unfortunately the tenant ID max length is way less because of memcached (we have the tenant ID as part of the cache key). Memcached max key length is 250 bytes. I will put a conservative 150 characters in the doc.